File: frontend/src/choices.js
export const implementation_status_choices = {
  PROJECT_NOT_STARTED: "Project not started",
  STARTUP_PHASE: "Start-up phase (no production)",
  IN_OPERATION: "In operation (production)",
  PROJECT_ABANDONED: "Project abandoned",
};

export const nature_of_deal_choices = {
  OUTRIGHT_PURCHASE: "Outright purchase",
  LEASE: "Lease",
  CONCESSION: "Concession",
  EXPLOITATION_PERMIT:
    "Exploitation permit / license / concession (for mineral resources)",
  PURE_CONTRACT_FARMING: "Pure contract farming",
};

export const negotiation_status_choices = {
  Intended: {
    EXPRESSION_OF_INTEREST: "Expression of interest",
    UNDER_NEGOTIATION: "Under negotiation",
    MEMORANDUM_OF_UNDERSTANDING: "Memorandum of understanding",
  },
  Concluded: {
    ORAL_AGREEMENT: "Oral agreement",
    CONTRACT_SIGNED: "Contract signed",
  },
  Failed: {
    NEGOTIATIONS_FAILED: "Negotiations failed",
    CONTRACT_CANCELED: "Contract canceled",
  },

  CONTRACT_EXPIRED: "Contract expired",
  CHANGE_OF_OWNERSHIP: "Change of ownership",
};

export const intention_of_investment_choices = {
  Agriculture: {
    BIOFUELS: "Biofuels",
    FOOD_CROPS: "Food crops",
    FODDER: "Fodder",
    LIVESTOCK: "Livestock",
    NON_FOOD_AGRICULTURE: "Non-food agricultural commodities",
    AGRICULTURE_UNSPECIFIED: "Agriculture unspecified",
  },
  Forestry: {
    TIMBER_PLANTATION: "Timber plantation",
    FOREST_LOGGING: "Forest logging / management",
    CARBON: "For carbon sequestration/REDD",
    FORESTRY_UNSPECIFIED: "Forestry unspecified",
  },

  Other: {
    MINING: "Mining",
    OIL_GAS_EXTRACTION: "Oil / Gas extraction",
    TOURISM: "Tourism",
    INDUSTRY: "Industry",
    CONVERSATION: "Conservation",
    LAND_SPECULATION: "Land speculation",
    RENEWABLE_ENERGY: "Renewable energy",
    OTHER: "Other",
  },
};

export const intention_of_investment_map = {
  BIOFUELS: ["Biofuels", "fas fa-leaf"],
  FOOD_CROPS: ["Food crops", "fas fa-carrot"],
  FODDER: ["Fodder", "fas fa-leaf"],
  LIVESTOCK: ["Livestock", "fas fa-paw"],
  NON_FOOD_AGRICULTURE: ["Non-food agricultural commodities", "fas fa-leaf"],
  AGRICULTURE_UNSPECIFIED: ["Agriculture unspecified", "fas fa-leaf"],

  TIMBER_PLANTATION: ["Timber plantation", "fas fa-tree"],
  FOREST_LOGGING: ["Forest logging / management", "fas fa-tree"],
  CARBON: ["For carbon sequestration/REDD", "fas fa-tree"],
  FORESTRY_UNSPECIFIED: ["Forestry unspecified", "fas fa-tree"],

  MINING: ["Mining", "fas fa-mountain"],
  OIL_GAS_EXTRACTION: ["Oil / Gas extraction", "fas fa-oil-can"],
  TOURISM: ["Tourism", "fas fa-plane"],
  INDUSTRY: ["Industry", "fas fa-industry"],
  CONVERSATION: ["Conservation", ""],
  LAND_SPECULATION: ["Land speculation", "fas fa-chart-line"],
  RENEWABLE_ENERGY: ["Renewable energy", "fas fa-wind"],
  OTHER: ["Other", ""],
};

export const classification_choices = {
  GOVERNMENT: "Government",
  GOVERNMENT_INSTITUTION: "Government institution",
  STATE_OWNED_COMPANY: "State-/government (owned) company",
  SEMI_STATE_OWNED_COMPANY: "Semi state-owned company",
  ASSET_MANAGEMENT_FIRM: "Asset management firm",
  BILATERAL_DEVELOPMENT_BANK:
    "Bilateral Development Bank / Development Finance Institution",
  STOCK_EXCHANGE_LISTED_COMPANY: "Stock-exchange listed company",
  COMMERCIAL_BANK: "Commercial Bank",
  INSURANCE_FIRM: "Insurance firm",
  INVESTMENT_BANK: "Investment Bank",
  INVESTMENT_FUND: "Investment fund",
  MULTILATERAL_DEVELOPMENT_BANK: "Multilateral Development Bank (MDB)",
  PRIVATE_COMPANY: "Private company",
  PRIVATE_EQUITY_FIRM: "Private equity firm",
  INDIVIDUAL_ENTREPRENEUR: "Individual entrepreneur",
  NON_PROFIT: "Non - Profit organization (e.g. Church, University etc.)",
  OTHER: "Other (please specify in comment field)",
};

File: apps/grid/forms/choices.py
"""
TODO: these are used in many DB queries! Move those to models and build
choices from them.
"""
from django.utils.translation import gettext_lazy as _

# Agriculture
INTENTION_BIOFUELS = "Biofuels"
INTENTION_FOOD_CROPS = "Food crops"
INTENTION_FODDER = "Fodder"
INTENTION_LIVESTOCK = "Livestock"
INTENTION_NON_FOOD_AGRI = "Non-food agricultural commodities"
INTENTION_AGRI_UNSPECIFIED = "Agriculture unspecified"
# Forestry
INTENTION_TIMBER_PLANTATION = "Timber plantation"
INTENTION_FOREST_LOGGING = "Forest logging / management"
INTENTION_CARBON = "For carbon sequestration/REDD"
INTENTION_FORESTRY_UNSPECIFIED = "Forestry unspecified"
# Other
INTENTION_MINING = "Mining"
INTENTION_OIL_GAS_EXTRACTION = "Oil / Gas extraction"
INTENTION_TOURISM = "Tourism"
INTENTION_INDUSTRY = "Industry"
INTENTION_CONVERSATION = "Conservation"
INTENTION_LAND_SPECULATION = "Land speculation"
INTENTION_RENEWABLE_ENERGY = "Renewable Energy"
INTENTION_OTHER = "Other"

intention_agriculture_choices = (
    (INTENTION_BIOFUELS, _("Biofuels")),
    (INTENTION_FOOD_CROPS, _("Food crops")),
    (INTENTION_FODDER, _("Fodder")),
    (INTENTION_LIVESTOCK, _("Livestock")),
    (INTENTION_NON_FOOD_AGRI, _("Non-food agricultural commodities")),
    (INTENTION_AGRI_UNSPECIFIED, _("Agriculture unspecified")),
)

INTENTION_AGRICULTURE_MAP = dict(intention_agriculture_choices)

intention_forestry_choices = (
    (INTENTION_TIMBER_PLANTATION, _("Timber plantation (for wood and fibre)")),
    (
        INTENTION_FOREST_LOGGING,
        _("Forest logging / management (for wood and fibre)"),
    ),  # new
    (INTENTION_CARBON, _("For carbon sequestration/REDD")),
    (INTENTION_FORESTRY_UNSPECIFIED, _("Forestry unspecified")),
)

INTENTION_FORESTRY_MAP = dict(intention_forestry_choices)

intention_other_choices = (
    (INTENTION_MINING, _("Mining")),
    (INTENTION_OIL_GAS_EXTRACTION, _("Oil / Gas extraction")),
    (INTENTION_TOURISM, _("Tourism")),
    (INTENTION_INDUSTRY, _("Industry")),
    (INTENTION_CONVERSATION, _("Conservation")),
    (INTENTION_LAND_SPECULATION, _("Land speculation")),
    (INTENTION_RENEWABLE_ENERGY, _("Renewable Energy")),
    (INTENTION_OTHER, _("Other (please specify)")),
)

intention_choices = (
    intention_agriculture_choices + intention_forestry_choices + intention_other_choices
)

INTENTION_MAP = dict(intention_choices)

grouped_intention_choices = (
    ("Agriculture", intention_agriculture_choices),
    ("Forestry", intention_forestry_choices),
    ("Other", intention_other_choices),
)

NATURE_OUTRIGHT_PURCHASE = "Outright purchase"
NATURE_LEASE = "Lease"
NATURE_CONCESSION = "Concession"
NATURE_EXPLOITATION_PERMIT = "Exploitation permit / license / concession"
NATURE_CONTRACT_FARMING = "Pure contract farming"
NATURE_OTHER = "Other"
nature_choices = (
    (NATURE_OUTRIGHT_PURCHASE, _("Outright purchase")),
    (NATURE_LEASE, _("Lease")),
    (NATURE_CONCESSION, _("Concession")),
    (
        NATURE_EXPLOITATION_PERMIT,
        _("Exploitation permit / license / concession (for mineral resources)"),
    ),
    (NATURE_CONTRACT_FARMING, _("Pure contract farming")),
    (NATURE_OTHER, _("Other")),
)

price_type_choices = (
    ("", _("---------")),
    ("per ha", _("per ha")),
    ("for specified area", _("for specified area")),
)

actor_choices = (
    ("", _("---------")),
    (
        "Government / State institutions",
        _(
            "Government / State institutions (government, ministries, departments, agencies etc.)"
        ),
    ),
    (
        "Traditional land-owners / communities",
        _("Traditional land-owners / communities"),
    ),
    (
        "Traditional local authority",
        _("Traditional local authority (e.g. Chiefdom council / Chiefs)"),
    ),
    ("Broker", _("Broker")),
    ("Intermediary", _("Intermediary")),
    ("Other", _("Other (please specify)")),
)
